apps/snitch_core/lib/core/data/model/order.ex

```diff
@@ -6,19 +6,118 @@ defmodule Snitch.Data.Model.Order do
   alias Snitch.Data.Schema.Order
   alias Snitch.Data.Model.LineItem, as: LineItemModel
 
-  @spec create(map, [map]) :: {:ok, Order.t()} | {:error, Ecto.Changeset.t()}
+  @doc """
+  Creates an order with supplied `params` and `line_items`.
+
+  Since line-items are "owned" by an order it does not make sense to persist
+  `LineItem`s first and then `create/2` an `Order`.
+
+  `params` is a map that is passed to the
+  `Snitch.Data.Schema.Order.changeset/3`.
+
+  > * `line_items` is not a list of `LineItem` schema structs, but just a list
+  >   of maps with the keys `:variant_id` and `:quantity`.
+  > * These `LineItem`s will be created (casted, to be precise) along with the
+  >   `Order` in a DB transaction.
+
+  ## Example
+  ```
+  line_items = [%{variant_id: 1, quantity: 42}, %{variant_id: 2, quantity: 42}]
+  params = %{user_id: 1}
+  {:ok, order} = Snitch.Data.Model.Order.create(params, line_items)
+  ```
+
+  ## See also
+  `Ecto.Changeset.cast_assoc/3`
+  """
+  @spec create(map, [line_items]) :: {:ok, Order.t()} | {:error, Ecto.Changeset.t()}
+        when line_items: %{variant_id: non_neg_integer, quantity: non_neg_integer}
   def create(params, line_items) do
-    priced_items = LineItemModel.update_price_and_totals(line_items)
+    priced_items = update_line_item_costs(line_items)
 
     QH.create(Order, Map.put(params, :line_items, priced_items), Repo)
   end
 
-  @spec update(map) :: {:ok, Order.t()} | {:error, Ecto.Changeset.t()}
-  def update(params) do
-    line_items = Map.get(params, :line_items, [])
-    priced_items = LineItemModel.update_price_and_totals(line_items)
+  @doc """
+  Updates the order with supplied `params`. `params` can include "new"
+  `line_items`.
+
+  ## Caution!
+
+  The `line_items` are "casted" with the order and if `params` does not include
+  a `line_items`, then **all previous line-items will be deleted!**
+
+  ### Retain previous `LineItem`s
+
+  If you wish to retain the line-items, you must pass a list of maps with the
+  line-item `:id`s, like so:
```

Comment:
Can we handle this on our end? Since we have the  `order id`, we pull it out from db  then update the list with params we received and not asking the user to explicitly enter `ids` of `lineitems` to be preserved. Again just a suggestion.